tests/test-sessions/test-sessions-common/src/main/java/org/eclipse/jetty/server/session/AbstractSessionDataStoreTest.java

```diff
@@ -205,37 +206,109 @@ public void testUpdateSession() throws Exception
     }
     
     
+    
+    
     /**
      * Test that the store can persist a session that contains 
-     * serializable objects in the attributes.
+     * serializable Proxy objects in the attributes.
      * 
      * @throws Exception
      */
-    /*
-     * @Test public void testStoreObjectAttributes() throws Exception { //create
-     * the SessionDataStore ServletContextHandler context = new
-     * ServletContextHandler(ServletContextHandler.SESSIONS);
-     * context.setContextPath("/test"); SessionDataStoreFactory factory =
-     * createSessionDataStoreFactory();
-     * ((AbstractSessionDataStoreFactory)factory).setGracePeriodSec(
-     * GRACE_PERIOD_SEC); SessionDataStore store =
-     * factory.getSessionDataStore(context.getSessionHandler()); SessionContext
-     * sessionContext = new SessionContext("foo", context.getServletContext());
-     * store.initialize(sessionContext);
-     * 
-     * store.start();
-     * 
-     * //create a session SessionData data = store.newSessionData("1234", 100,
-     * 200, 199, -1);//never expires TestFoo testFoo = new TestFoo();
-     * testFoo.setInt(33); FooInvocationHandler handler = new
-     * FooInvocationHandler(testFoo); Foo foo =
-     * (Foo)Proxy.newProxyInstance(Thread.currentThread().getContextClassLoader(
-     * ), new Class[] {Foo.class}, handler); data.setAttribute("foo", foo);
-     * data.setLastNode(sessionContext.getWorkerName());
-     * 
-     * //test that it can be persisted store.store("1234", data);
-     * checkSessionPersisted(data); }
-     */
+    @Test 
+    public void testStoreObjectAttributes() throws Exception 
+    {
+        
+        //Use classes that would only be known to the webapp classloader
+        InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("Proxyable.clazz");
+        File proxyabledir = new File (MavenTestingUtils.getTargetDir(), "proxyable");
+        proxyabledir.mkdirs();
+        File proxyableClass = new File (proxyabledir, "Proxyable.class");
+        IO.copy(is, new FileOutputStream(proxyableClass));
+        is.close();
+        
+        assertTrue(proxyableClass.exists());
+        assertTrue(proxyableClass.length() != 0);
+        
+        is =  Thread.currentThread().getContextClassLoader().getResourceAsStream("ProxyableInvocationHandler.clazz");
+        File pihClass = new File (proxyabledir, "ProxyableInvocationHandler.class");
+        IO.copy(is, new FileOutputStream(pihClass));
+        is.close();
```

Comment:
Why precompile these classes like this?  
Having the *.java files in `src/main/resources` seems odd too.

If they exist in `src/main/java`, and we allow normal compilation, why not just move then to `proxyabledir`?

Reply:
Done this way because anything you put in src/main/java will be compiled and put onto the classpath at runtime: we don't want these classes on the classpath at runtime at all, only in a place that would be the equivalent of an unpacked webapp.